oauth2client/contrib/gce.py

```diff
@@ -123,3 +127,60 @@ def create_scoped_required(self):
 
     def create_scoped(self, scopes):
         return AppAssertionCredentials(scopes, **self.kwargs)
+
+    def sign_blob(self, blob):
+        """Cryptographically sign a blob (of bytes).
+
+        This method is provided to support a common interface, but
+        the actual key used for a Google Compute Engine service account
+        is not available, so it can't be used to sign content.
+
+        Args:
+            blob: bytes, Message to be signed.
+
+        Raises:
+            NotImplementedError, always.
```

Comment:
The desire is for `credentials.sign_blob(blob)` to be possible. Essentially it's choosing between `NotImplementedError` and `AttributeError`. Which duck is best?

The reason I did it this way was so that users could get a helpful message

```
Compute Engine service accounts cannot sign blobs
```

rather than just an `AttributeError`. (In `gcloud-python` we have a [note][1] about this in our docstring.)

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/a77e7fae62237edbba25b5da37deb673f6ef92b5/gcloud/storage/blob.py#L158-L166